### PR TITLE
refactor: optimize the process of graph rag

### PR DESCRIPTION
**Pull request by ChenZiHong-Gavin**
- [x] replace print with log
- [x] delete single quote in keywords
- [x] special handling is required when keywords list is empty
- [ ] regarding the issue of duplicate vid attributes: compare the effectiveness of soltions
- [ ] null value detection on vid attributes to reduce unnecessary token consumption
- [ ] add few shot on KEYWORDS EXTRACTION and KEYWORDS EXPAND prompt template

### CODE REVIEW

**Comment by ChenZiHong-Gavin** [comment]
image
As can be seen in fig, single quotes may lead to gremlin error. (however, double quotes won't)

File: hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/graph_rag_query.py [hunk 1]
```diff
@@ -229,15 +230,20 @@ def _process_path(self, path: Any, use_id_to_match: bool) -> Tuple[str, List[str
...
         if matched_str in node_cache:
             flat_rel = flat_rel[:-prior_edge_str_len]
             return flat_rel, prior_edge_str_len, depth
 
         node_cache.add(matched_str)
         props_str = ", ".join(f"{k}: {v}" for k, v in item["props"].items())
-        node_str = f"{item['id']}{{{props_str}}}"
+        # TODO: we may remove label id or replace with label name
+        if matched_str in global_vertex_cache:
+            node_str = f"{item['id']}"
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/graph_rag_query.py:243]:
```suggestion
            node_str = matched_str
```

File: hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/graph_rag_query.py [hunk 2]
```diff
@@ -192,18 +191,19 @@ def _format_graph_query_result(self, query_paths) -> Tuple[Set[str], List[Set[st
         subgraph = set()
         subgraph_with_degree = {}
         vertex_degree_list: List[Set[str]] = []
+        global_vertex_cache: Set[str] = set()
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/graph_rag_query.py:194]: use a short name like `v_cache`? same as `e_cache`

Comment (ChenZiHong-Gavin) [reply hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/graph_rag_query.py:194]: > use a short name like `v_cache` or `v_set`? same as `e_cache`

`v_cache` and `e_cache` seems good

**Review by imbajin — APPROVED** [review]

**Pull request merged**